File: aat-android/src/main/java/ch/bailu/aat/description/AltitudeDescription.java
package ch.bailu.aat.description;

import android.content.Context;

import java.text.DecimalFormat;

import ch.bailu.aat.preferences.Storage;
import ch.bailu.aat_lib.gpx.GpxInformation;
import ch.bailu.aat_lib.preferences.general.SolidUnit;
import ch.bailu.aat_lib.resources.Res;

public class AltitudeDescription extends FloatDescription {

    private final SolidUnit unit;


    public AltitudeDescription(Context context) {
        unit = new SolidUnit(new Storage(context));
    }



    @Override
    public String getLabel() {
        return Res.str().altitude();
    }

    @Override
    public String getUnit() {
        return unit.getAltitudeUnit();
    }

    public String getValue() {
        return getValue(getCache());
    }

    private static final DecimalFormat f0 = new DecimalFormat("0");

    public String getValue(float v) {
        float f = unit.getAltitudeFactor();
        return f0.format(v *f);
    }


    public String getValueUnit(float v) {
        return getValue(v) + " " + getUnit();
    }

    @Override
    public void onContentUpdated(int iid, GpxInformation info) {
        setCache( ((float)info.getAltitude()) );
    }

}

File: aat-lib/src/main/java/ch/bailu/aat_lib/description/AltitudeDescription.kt
package ch.bailu.aat_lib.description

import ch.bailu.aat_lib.gpx.information.GpxInformation
import ch.bailu.aat_lib.preferences.StorageInterface
import ch.bailu.aat_lib.preferences.general.SolidUnit
import ch.bailu.aat_lib.resources.Res

open class AltitudeDescription(storageInterface: StorageInterface) : FloatDescription() {
    private val unit = SolidUnit(storageInterface)
    override fun getLabel(): String {
        return Res.str().altitude()
    }

    override fun getUnit(): String {
        return unit.altitudeUnit
    }

    override fun getValue(): String {
        return getValue(cache)
    }

    fun getValue(value: Float): String {
        return FormatDisplay.f().decimal1.format(value * unit.altitudeFactor)
    }

    fun getValueUnit(v: Float): String {
        return getValue(v) + " " + getUnit()
    }

    override fun onContentUpdated(iid: Int, info: GpxInformation) {
        setCache(info.getAltitude())
    }

    fun getAltitudeDescription(value: Float): String {
        return getValue(value) + " " + unit.altitudeUnit
    }
}
